frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Badge } from '@patternfly/react-core';
+import * as classNames from 'classnames';
+import '../_resource.scss';
+
+/**
+ * Component for displaying resource status badge.
+ * Use this component to display status of given resource.
+ * It accepts child element to be rendered inside the badge.
+ * @component ResourceStatus
+ * @example
+ * ```ts
+ * return (
+ *  <ResourceStatus additionalClassNames="hidden-xs">
+ *    <Status status={resourceStatus} />
+ *  </ResourceStatus>
+ * )
+ * ```
+ */
+const ResourceStatus: React.FC<ResourceStatusProps> = ({
+  additionalClassNames,
+  badgeAlt,
+  children,
+}) => {
+  return (
+    <span className={classNames('dps-resource-item__resource-status', additionalClassNames)}>
+      <Badge
+        className={classNames('dps-resource-item__resource-status-badge', {
+          'dps-resource-item__resource-status-badge--alt': badgeAlt,
+        })}
+        isRead
+        data-test="resource-status"
+      >
+        {children}
+      </Badge>
+    </span>
+  );
+};
+
+export type ResourceStatusProps = {
```

Comment:
No need to export the props of a component -- also could we have this at the top of the file too -- I think it's easier to read then putting it at the bottom?

```
// outside of this file, you can infer the props type from the component
type ResourceStatusProps = React.ComponentProps<typeof ResourceStatus>;